### PR TITLE
Allow library to parse ISO 8601 dates, in addition to existing one.

### PR DESCRIPTION
**Pull request by noordawod**
https://en.wikipedia.org/wiki/ISO_8601

The library will now support both ISO 8601 and RFC 822 dates.

### CODE REVIEW

**Comment by einmalfel** [comment]
thanks for contribution, @noordawod ! PR looks cool, will merge it soon. Could you please provide a link to feed that uses ISO 8601 date formatting, so I'll be able to make a test for this feature?

**Comment by noordawod** [comment]
Right on it, @einmalfel. Since I mostly trusted my judgement on the matter, let me perform this test myself and submit any more patches (if needed) and then I'll send you a feed.

**Comment by noordawod** [comment]
@einmalfel 

Okay, now I've made sure the patch works and even widened the coverage of `Utils.parseDate()` so that it's the only function being used to parse dates. Why? Well, because developers don't follow standards unfortunately and we have to try few formats to parse the date before giving up...

Anyway, now the patch is complete and tested against few feeds.

Here's one for testing: http://feeds.feedburner.com/AniaMaluje-BlogUrodowy

By the way, we're using your library to cover our Atom/RSS needs in an upcoming mobile app we're working on, so I'll probably submit more pulls as time goes by. Thanks for the great work :smile:

**Comment by noordawod** [comment]
@einmalfel 

You think you could take a look at the two pulls today? Would be nice :smile: 

Thanks!

**Comment by einmalfel** [comment]
@noordawod 
Sorry about the delay! I'll check & merge PRs in the evening. I wont have a chance to look into new code until then :(

**Comment by noordawod** [comment]
Sounds good, @einmalfel. I tested both obviously but would be nice to have another set of eyes take a look.

File: earl/src/main/java/com/einmalfel/earl/Utils.java [hunk 1]
```diff
@@ -29,17 +29,88 @@
...
     try {
       return rfc822DateTimeFormat.parse(dateString);
     } catch (ParseException exception) {
-      Log.w(TAG, "Malformed date " + dateString);
       return null;
     }
   }
 
+  @Nullable
+  private static Date parseRFC8601Date(@NonNull String dateString) {
```
Comment (einmalfel) [inline earl/src/main/java/com/einmalfel/earl/Utils.java:68]: If I understand correctly, it's ISO 8601, not RFC

Comment (noordawod) [reply earl/src/main/java/com/einmalfel/earl/Utils.java:68]: Correct. Did you change the code or should I do it?

**Comment by einmalfel** [comment]
I'll change it

**Pull request merged**